components/service/glean/src/main/java/mozilla/components/service/glean/config/Configuration.kt

```diff
@@ -37,6 +36,12 @@ data class Configuration @JvmOverloads constructor (
      * @return a [mozilla.telemetry.glean.config.Configuration] instance.
      */
     fun toWrappedConfiguration(): GleanCoreConfiguration {
-        return GleanCoreConfiguration(serverEndpoint, channel, maxEvents, httpClient)
+        // Only pass on the HTTP client if one was configured,
+        // otherwise fall back to the default in Glean core.
+        if (httpClient != null) {
+            return GleanCoreConfiguration(serverEndpoint, channel, maxEvents, httpClient)
+        } else {
+            return GleanCoreConfiguration(serverEndpoint, channel, maxEvents)
```

Comment:
Is our default behaving the same as `lib-fetch-httpurlconnection`? Please note that this will have impact on Lockwise and Firefox Reality. I'd rather making this explicit..

Reply:
Make it explicit how?
We don't want to depend on `lib-fetch-httpurlconnection`.
We implement a ping uploader here: https://github.com/mozilla/glean/blob/4aab82a58c0e9db1577ebd10019447155bb846e7/glean-core/android/src/main/java/mozilla/telemetry/glean/net/HttpURLConnectionUploader.kt#L20
It's defering upload to `java.net.HttpURLConnection`.

IMO it makes more sense to _not_ redefine anything like that in a-c.
That only means we would now introduce a second place we need to change if we ever touch the Glean core uploader.

Reply:
Sorry I wasn't clear :-) I mean let's just drop any default and force users to explicitly state what they want.

Reply:
If we only drop it here than that configuration API differs between a-c and direct use, which seems ... not good

Reply:
The API already differs between a-c and direct use: a-c depends on `ConceptFetch` while direct use doesn't know about it.

If the intent here is to make sure Mozilla users are intentional with the upload mechanism, I'm afraid this is the only option.

Reply:
That would mean we're also breaking all the current users, as they are now forced to pass this explicitly.
Are we okay with doing that?

Reply:
> Are we okay with doing that?

This is what @pocmo [was asking](https://github.com/mozilla-mobile/android-components/issues/6660#issuecomment-624045948): I'm not a fan of this dance, to be honest, but we're living in A-C and we should respect the A-C's guidelines :-)

Reply:
> but we're living in A-C and we should respect the A-C's guidelines :-)

I'm open to discussing this though. :)

I think it is a fairly cheap breakage: Apps can fix this with one line of code. Fenix shouldn't break since Fenix should never use anything but Gecko for network communication. If it breaks Fenix then this would be good. :)

Lockwise: Yeah, they would probably need to add `HttpUrlConnectionClient` manually once. I think that is fine.

Firefox Reality: Haven't looked at it yet. But if it breaks then this may actually be good since I assume there is a strong reason to also use Gecko for all network communication in Firefox Reality. And if they have been using `HttpUrlConnection` instead then this may exactly be one of the cases where the default was not obvious (or having the choice) and where the default is the opposite of what we wanted.

Reply:
Firefox Reality seems to use the default and I filed an issue for changing that: https://github.com/MozillaReality/FirefoxReality/issues/3339